overlord/devicestate/devicestate.go

```diff
@@ -1239,6 +1239,10 @@ func Remodel(st *state.State, new *asserts.Model, localSnaps []*snap.SideInfo, p
 		return nil, fmt.Errorf("cannot remodel from grade %v to grade %v", current.Grade(), new.Grade())
 	}
 
+	if new.Base() == "" && current.Base() != "" {
+		return nil, errors.New("cannot remodel from snapd based system to core based system")
```

Comment:
we don't seem to use this terminogy in snapstate:

maybe "cannot remodel from UC18+ (using snapd snap) system back to UC16 system (using core snap)"?